tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_axis_utils.ts

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {Formatter} from '../lib/public_types';
+import {LinearScale} from '../lib/scale';
+
+export interface MinorTick {
+  value: number;
+  tickFormattedString: string;
+}
+
+// Major tick, unlike the minor tick, spans a range.
+export interface MajorTick {
+  // Start of the major tick range. An end is implicitly defined by next major tick while
+  // it can certainly change to explicitly define the end.
+  start: number;
+  tickFormattedString: string;
+}
+
+function getNumLeadingZerosInFractional(value: number): number {
+  const maybeExponential = value.toExponential().split('e-', 2);
+  if (maybeExponential.length === 2) {
+    return Number(maybeExponential[1]) - 1;
+  }
+  return 0;
+}
+
+export function getTicksForLinearScale(
+  scale: LinearScale,
+  formatter: Formatter,
+  maxMinorTickCount: number,
+  low: number,
```

Comment:
For consistency, would it make sense to just pass type `range: [number, number]`. Seems like this is a common type in your Scale and Axis code.